### ISSUE TITLE
A patient is waiting in a queue for a destroyed surgery

### ISSUE THREAD

**Issue by l29ah** [CorsixTH/CorsixTH#1208]
https://dump.bitcheese.net/files/turugyf/destroyed-queue.sav
ea0e6df, Gentoo Linux, multiple `Warning: Proper idle not in action_queue` in the log.

**Comment by mugmuggy** [comment]
I can't get this save to load but there appears it could be related to one https://github.com/CorsixTH/CorsixTH/issues/444#issuecomment-67879296 with a similar outcome, And I have happened on the possibility in some other saves as well.

When a patient is in a queue, and they say for instance they are about to access the drinks machine and the room is destroyed (even with editing as in #444), the Queue:rerouteAllPatients call doesn't interrupt the drinks machine actions and thus finishes whatever action is currently runnning and continue with the rest of the actions. Leaving the patient to rejoin the queue with an idle and seek room action afterwards.

Patient:tickDay handles a similar issue already:
https://github.com/CorsixTH/CorsixTH/blob/9de46b728c61fedf96cd1120a8040af328a55fa8/CorsixTH/Lua/entities/patient.lua#L909-L936

For my particular case I added a check for the destination room and its queue existing which sent them on to the reception desk. The 444 case, actually has the action referring to the old queue and the next room is valid with a valid queue. Additional checks to ensure the queue action queue and the next room queue is also the same queue should fix the 444 case, but I haven't yet reproduced the circumstances that triggered that one.
